### PR TITLE
Add structured logging

### PR DESCRIPTION
**Pull request by kakkoyun**
This PR introduces structured logging and removes usage of the standard logger to improve troubleshooting experience.

### CODE REVIEW

**Comment by kakkoyun** [comment]
cc @metalmatze

File: cmd/telemeter-benchmark/main.go [hunk 1]
```diff
@@ -120,14 +139,15 @@ func runCmd() error {
...
+		Logger:      opt.Logger,
 	}
 
 	b, err := benchmark.New(cfg)
 	if err != nil {
 		return fmt.Errorf("failed to configure the Telemeter benchmarking tool: %v", err)
 	}
 
-	log.Printf("Starting telemeter-benchmark against %s (listen=%s)", opt.To, opt.Listen)
+	level.Info(opt.Logger).Log("msg", fmt.Sprintf("Starting telemeter-benchmark against %s (listen=%s)", opt.To, opt.Listen))
```
Comment (metalmatze) [inline cmd/telemeter-benchmark/main.go]: Using sprintf in combination with structured logging should be a red flag. :slightly_smiling_face: 

```suggestion
	level.Info(opt.Logger).Log("msg", "starting telemeter-benchmark", "to", opt.To, "addr", opt.Listen))
```

File: cmd/telemeter-benchmark/main.go [hunk 2]
```diff
@@ -153,11 +173,11 @@ func runCmd() error {
 				select {
 				case <-hup:
 					if err := b.Reconfigure(cfg); err != nil {
-						log.Printf("error: failed to reload config: %v", err)
+						level.Error(opt.Logger).Log("msg", fmt.Sprintf("failed to reload config: %v", err))
```
Comment (metalmatze) [inline cmd/telemeter-benchmark/main.go]: The most canonical would be to have a KV pair with `"msg", ..., "err", ...`
```suggestion
						level.Error(opt.Logger).Log("msg", "failed to reload config", "err", err))
```

**Comment by kakkoyun** [comment]
cc @squat since you're back, I'd love to hear your opinion as well

File: cmd/telemeter-benchmark/main.go [hunk 3]
```diff
@@ -62,7 +69,21 @@ func main() {
...
+	cmd.Flags().StringVar(&opt.LogLevel, "log-level", opt.LogLevel, "Log filtering level. e.g info, debug, warn, error")
+
+	l := log.NewLogfmtLogger(log.NewSyncWriter(os.Stderr))
+	lvl, err := cmd.Flags().GetString("log-level")
+	if err != nil {
+		level.Error(l).Log("msg", "could not parse log-level.")
+	}
+	l = level.NewFilter(l, logger.LogLevelFromString(lvl))
+	l = log.WithPrefix(l, "ts", log.DefaultTimestampUTC)
+	l = log.WithPrefix(l, "caller", log.Caller(3))
```
Comment (brancz) [inline cmd/telemeter-benchmark/main.go]: Why not `log.DefaultCaller`? I just went through the code just to find it's the same value, as I thought this was something special.

File: cmd/telemeter-client/main.go [hunk 4]
```diff
@@ -65,7 +69,22 @@ func main() {
...
+	cmd.Flags().StringVar(&opt.LogLevel, "log-level", opt.LogLevel, "Log filtering level. e.g info, debug, warn, error")
+
+	l := log.NewLogfmtLogger(log.NewSyncWriter(os.Stderr))
+	lvl, err := cmd.Flags().GetString("log-level")
+	if err != nil {
+		level.Error(l).Log("msg", "could not parse log-level.")
+	}
+	l = level.NewFilter(l, logger.LogLevelFromString(lvl))
+	l = log.WithPrefix(l, "ts", log.DefaultTimestampUTC)
+	l = log.WithPrefix(l, "caller", log.Caller(3))
```
Comment (brancz) [inline cmd/telemeter-client/main.go]: same

File: cmd/telemeter-server/main.go [hunk 5]
```diff
@@ -129,7 +134,22 @@ func main() {
...
+	cmd.Flags().StringVar(&opt.LogLevel, "log-level", opt.LogLevel, "Log filtering level. e.g info, debug, warn, error")
+
+	l := log.NewLogfmtLogger(log.NewSyncWriter(os.Stderr))
+	lvl, err := cmd.Flags().GetString("log-level")
+	if err != nil {
+		level.Error(l).Log("msg", "could not parse log-level.")
+	}
+	l = level.NewFilter(l, logger.LogLevelFromString(lvl))
+	l = log.WithPrefix(l, "ts", log.DefaultTimestampUTC)
+	l = log.WithPrefix(l, "caller", log.Caller(3))
```
Comment (brancz) [inline cmd/telemeter-server/main.go]: same

File: cmd/telemeter-benchmark/main.go (hunk 3 above)
Comment (kakkoyun) [reply cmd/telemeter-benchmark/main.go]: Yes, it was the default one. I just have forgotten to switch it back after several experiments. I'll fix it right away.

**Pull request merged**